test/spec/exception-handling.wast

```diff
@@ -2,38 +2,110 @@
   (event $e0 (attr 0) (param i32))
   (event $e1 (attr 0) (param i32 f32))
 
-  (func (export "throw_test")
-    (throw $e0 (i32.const 0))
+  (func (export "throw_single_value")
+    (throw $e0 (i32.const 5))
   )
 
-  (func (export "rethrow_test")
+  (func (export "throw_multiple_values")
+    (throw $e1 (i32.const 3) (f32.const 3.5))
+  )
+
+  (func (export "rethrow_nullref")
     (rethrow (ref.null))
   )
 
-  (func (export "try_test") (result i32)
+  (func (export "try_nothrow") (result i32)
     (try (result i32)
       (i32.const 3)
+      (catch
+        (drop (exnref.pop))
+        (i32.const 0)
+      )
+    )
+  )
+
+  (func (export "try_throw_catch") (result i32)
+    (try (result i32)
+      (throw $e0 (i32.const 5))
       (catch
         (drop (exnref.pop))
         (i32.const 3)
       )
     )
   )
 
-  (func (export "br_on_exn_test") (result i32)
+  (func (export "try_throw_rethrow")
+    (try
+      (throw $e0 (i32.const 5))
+      (catch
+        (rethrow (exnref.pop))
+      )
+    )
+  )
+
+  (func (export "br_on_exn_nullref") (result i32)
     (block $l0 (result i32)
       (drop
         (br_on_exn $l0 $e0 (ref.null))
       )
       (i32.const 0)
     )
   )
+
+  (func (export "br_on_exn_match_single_value") (result i32) (local $exn exnref)
+    (try (result i32)
+      (throw $e0 (i32.const 5))
+      (catch
+        (local.set $exn (exnref.pop))
+        (block $l0 (result i32)
+          (rethrow
+            (br_on_exn $l0 $e0 (local.get $exn))
+          )
+        )
+      )
+    )
+  )
+
+  (func (export "br_on_exn_match_multiple_values") (result i32 f32)
```

Comment:
Should this also work with zero values?

Reply:
Yes. Added `br_on_exn_match_no_values` test for that.